### DISCUSSION TITLE
high availability / clustering

### DISCUSSION THREAD

**Discussion by jousby** [geldata/gel#2683] [Ideas]
Hi, 

Are there any plans to make the edgedb process that sits in front of postgres clustered/multi node? I saw the RFC discussion around postgres failover but curious to see if there will be plans to run the edgedb process in an active/active or active/passive setup. 

My wishlist would include a quickstart for running on AWS in a way that is at least autohealing. Also curious to hear if there plans for autoscaling?

**Comment by elprans** [comment]
> Are there any plans to make the edgedb process that sits in front of postgres clustered/multi node?

Yes! We are working on this as I'm writing this.  See, for example, https://github.com/edgedb/edgedb/pull/2574 and https://github.com/edgedb/edgedb/pull/2580

The EdgeDB server itself is stateless, and so is easy have HA for it: you simply need to run multiple copies in front of the same Postgres cluster.

> Also curious to hear if there plans for autoscaling?

EdgeDB scalability is mostly a function of the scalability of the underlying PostgreSQL cluster.  We are working on adding support for automatic traffic steering to read replicas and so EdgeDB will just work with any Postgres auto-scaler that supports replica discovery in a reasonable way.  We'll likely add explicit support for Amazon Aurora Auto Scaling as well as popular open-source solutions like Stolon and Patroni.

**Reply by jousby** [reply]
Nice!

And the traffic in front of those stateless instances.... I saw that there is a EdgeQL and GraphQL over HTTP option. Sounds like you can just round robin that via a traditional layer 7 load balancer like Application Load Balancer on AWS. What about the language specific driver options that support transactions and ddl though? Whats the transport / ports for those (apologies if i missed this in the docs somewhere). I saw in the PR that you are looking at options like Consul to service discovery but I'm also wondering if that traffic is not HTTP based if you can use something like a Network Load Balancer (NLB) on AWS.
